From ed9e3619b799f6744ae11cf919a72eb5eb82e1d8 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 24 Apr 2017 14:38:06 +0200
Subject: [PATCH 10/62] JsonSchemaBuilder/Makefile: build out-of-tree

Don't need a copy of src/ for this.  This was fragile anyway.

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 tools/depends/native/JsonSchemaBuilder/.gitignore |  5 +++++
 tools/depends/native/JsonSchemaBuilder/Makefile   | 14 ++++++--------
 2 files changed, 11 insertions(+), 8 deletions(-)
 create mode 100644 tools/depends/native/JsonSchemaBuilder/.gitignore

diff --git a/tools/depends/native/JsonSchemaBuilder/.gitignore b/tools/depends/native/JsonSchemaBuilder/.gitignore
new file mode 100644
index 0000000..bbd8cbf
--- /dev/null
+++ b/tools/depends/native/JsonSchemaBuilder/.gitignore
@@ -0,0 +1,5 @@
+/src/Makefile.in
+/src/aclocal.m4
+/src/*.cache
+/src/config/
+/src/configure
diff --git a/tools/depends/native/JsonSchemaBuilder/Makefile b/tools/depends/native/JsonSchemaBuilder/Makefile
index 3abd729..4c87012 100644
--- a/tools/depends/native/JsonSchemaBuilder/Makefile
+++ b/tools/depends/native/JsonSchemaBuilder/Makefile
@@ -15,19 +15,17 @@ endif
 
 SOURCE=$(ROOT_DIR)/src
 
-CONFIGURE=./configure --prefix=$(PREFIX)
+CONFIGURE=$(SOURCE)/configure --prefix=$(PREFIX)
 APP=$(PREFIX)/bin/JsonSchemaBuilder
 
 all: $(APP)
 
-$(PLATFORM)/autogen.sh: $(SOURCE)/*
-	rm -rf $(PLATFORM)
-	cp -a $(SOURCE) $(PLATFORM)
-
-$(PLATFORM)/configure: $(PLATFORM)/autogen.sh
-	cd $(PLATFORM) && ./autogen.sh
+$(SOURCE)/configure: $(SOURCE)/configure.ac
+	cd $(SOURCE) && ./autogen.sh
 
-$(PLATFORM)/config.log: $(PLATFORM)/configure
+$(PLATFORM)/config.log: $(SOURCE)/configure
+	rm -rf $(PLATFORM)
+	mkdir -p $(PLATFORM)
 	cd $(PLATFORM) && $(CONFIGURE)
 
 $(PLATFORM)/JsonSchemaBuilder: $(PLATFORM)/config.log
-- 
2.7.4

